corehq/feature_previews.py

```diff
@@ -133,3 +158,39 @@ def enable_callcenter(domain_name, checked):
     ),
     privilege=LOOKUP_TABLES,
 )
+
+
+def is_eligible_for_ecd_preview(request):
+    if not (hasattr(request, 'plan')
+            and hasattr(request, 'subscription')
+            and hasattr(request, 'domain')):
+        return False
+
+    if request.subscription.is_trial:
+        return False
+
+    is_migrated = ECD_MIGRATED_DOMAINS.enabled_for_request(request)
+    is_enterprise_eligible = ECD_PREVIEW_ENTERPRISE_DOMAINS.enabled_for_request(request)
+    is_pro_or_advanced = request.plan.plan.edition in [
+        SoftwarePlanEdition.ADVANCED,
+        SoftwarePlanEdition.PRO
+    ]
+
+    return is_migrated and (is_pro_or_advanced or is_enterprise_eligible)
+
+
+def save_ecd_preview(domain_name, _checked):
```

Comment:
nit: this isn't very descriptive of what this function actually does - maybe something like `clear_dropdown_cache`?

Reply:
updated. thanks!